### PR TITLE
Using new code create API

### LINKED ISSUE #393: Migration ComputationalResourceWidget to AiiDA 2.1
See details:
https://github.com/aiidateam/aiida-core/blob/main/CHANGELOG.md#improved-interface-for-creating-codes

Here's the deprecation warnings that I am seeing:

```
/opt/conda/lib/python3.9/site-packages/aiida/orm/nodes/data/code/legacy.py:23: AiidaDeprecationWarning: The `Code` class is deprecated. To create an instance, use the `aiida.orm.nodes.data.code.installed.InstalledCode` or `aiida.orm.nodes.data.code.portable.PortableCode` for a "remote" or "local" code, respectively. If you are using this class to compare type, e.g. in `isinstance`, use `aiida.orm.nodes.data.code.abstract.AbstractCode`. (this will be removed in v3)
  warn_deprecation(
/opt/conda/lib/python3.9/site-packages/aiida/orm/utils/builders/code.py:19: AiidaDeprecationWarning: This module is deprecated. To create a new code instance, simply use the constructor. (this will be removed in v3)
  warn_deprecation('This module is deprecated. To create a new code instance, simply use the constructor.', version=3)
```

I am not an expert on this widget and have limited means of testing it, so leaving this one up to you @yakutovicha or @unkcpz.

Note that this functionality seems to come from version 2.1. But since the AiiDAlab Docker image is on this version already anyway, we might as well skip to support version 2.1 and don't support 2.0. (The current Docker images needs more fixes, so we never had a proper support for 2.0 anyway)

### PR DESCRIPTION
**Pull request by unkcpz**
Closes #393

### CODE REVIEW

**Comment by unkcpz** [comment]
This will still have one `Code` deprecation warning but rooted from aiida-core as mentioned by @sphuber today, I think we can not do anything from the AiiDAlab side.

File: aiidalab_widgets_base/computational_resources.py [hunk 1]
```diff
@@ -1059,23 +1058,24 @@ def on_setup_code(self, _=None):
...
-            kwargs["code_type"] = CodeBuilder.CodeType.ON_COMPUTER
 
             # Checking if the code with this name already exists
             qb = orm.QueryBuilder()
             qb.append(
                 orm.Computer, filters={"uuid": kwargs["computer"].uuid}, tag="computer"
             )
             qb.append(
-                orm.Code, with_computer="computer", filters={"label": kwargs["label"]}
+                orm.AbstractCode,
```
Comment (danielhollas) [inline aiidalab_widgets_base/computational_resources.py:1076]: Should this be `InstalledCode`?  InstalledCode is used on line 1084, so currently the code seems a bit inconsistent.

**Comment by danielhollas** [comment]
Looking at the error, it looks like the distutils issue that @unkcpz mentioned on Slack, and that is explained in aiida-core as well

**Comment by unkcpz** [comment]
> I am so annoyed by those random readthedocs failures.

As @danielhollas said, yes, I think this will be fixed by pining the `setuptools` version, but we are going to move to `pyproject.toml` and we 'll do it after that. So maybe we can ignore the failed DOC build failure at the moment?

File: aiidalab_widgets_base/computational_resources.py (hunk 1 above)
Comment (unkcpz) [inline aiidalab_widgets_base/computational_resources.py:1076]:
```suggestion
                orm.InstalledCode,
```

**Review by danielhollas — APPROVED** [review]
LGTM. I didn't actually test the changes as it is kind of hard for me to do so without access to Swiss clusters but I trust that you did :-)

**Pull request merged**